templates/go/client.go.twig

```diff
@@ -1,21 +1,97 @@
 package {{ spec.title | caseLower }}
 
 import (
+	"bytes"
+	"crypto/tls"
 	"encoding/json"
-  	"io/ioutil"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
 	"net/http"
+	"net/http/cookiejar"
 	"net/url"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strconv"
 	"strings"
+	"time"
 )
 
-// Client is the client struct to access {{ spec.title }} services
+const (
+	fileNameKey = "file"
+)
+
+// {{ spec.title | caseUcfirst }}Error represents an error of a client request
+type {{ spec.title | caseUcfirst }}Error struct {
+	statusCode int
+	message    string
+}
+
+// ClientResponse - represents the client response
+type ClientResponse struct {
+	Status     string
+	StatusCode int
+	Header     http.Header
+	Result     string
+}
+
+func (ce *{{ spec.title | caseUcfirst }}Error) Error() string {
+	return ce.message
+}
+
+func (ce *{{ spec.title | caseUcfirst }}Error) GetMessage() string {
+	return ce.message
+}
+
+func (ce *{{ spec.title | caseUcfirst }}Error) GetStatusCode() int {
+	return ce.statusCode
+}
+
+// Client is the client struct to access {{ spec.title | caseUcfirst }}  services
 type Client struct {
 	client     *http.Client
-	endpoint   string
 	headers    map[string]string
+	endpoint   string
+	timeout    time.Duration
 	selfSigned bool
 }
 
+// NewClient initializes a new {{ spec.title | caseUcfirst }} client with a given timeout
+func NewClient(timeout time.Duration) Client {
```

Comment:
Can we make the timeout optional? is is possible to keep the signature consistent?